sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubOptions.cs

```diff
@@ -0,0 +1,407 @@
+
+        private static string EscapeStorageCharacter(char character)
+        {
+            var ordinalValue = (ushort)character;
+            if (ordinalValue < 0x100)
+            {
+                return string.Format(CultureInfo.InvariantCulture, ":{0:X2}", ordinalValue);
+            }
+            else
+            {
+                return string.Format(CultureInfo.InvariantCulture, "::{0:X4}", ordinalValue);
+            }
+        }
+
+        // Escape a blob path.
+        // For diagnostics, we want human-readble strings that resemble the input.
+        // Inputs are most commonly alphanumeric with a fex extra chars (dash, underscore, dot).
+        // Escape character is a ':', which is also escaped.
+        // Blob names are case sensitive; whereas input is case insensitive, so normalize to lower.
+        private static string EscapeBlobPath(string path)
+        {
+            StringBuilder sb = new StringBuilder(path.Length);
+            foreach (char c in path)
+            {
+                if (c >= 'a' && c <= 'z')
+                {
+                    sb.Append(c);
+                }
+                else if (c == '-' || c == '_' || c == '.')
+                {
+                    // Potentially common carahcters.
+                    sb.Append(c);
+                }
+                else if (c >= 'A' && c <= 'Z')
+                {
+                    sb.Append((char)(c - 'A' + 'a')); // ToLower
+                }
+                else if (c >= '0' && c <= '9')
+                {
+                    sb.Append(c);
+                }
+                else
+                {
+                    sb.Append(EscapeStorageCharacter(c));
+                }
+            }
+
+            return sb.ToString();
+        }
+
+        internal static string GetEventHubNamespace(EventHubsConnectionStringBuilder connectionString)
+        {
+            // EventHubs only have 1 endpoint.
+            var url = connectionString.Endpoint;
+            var @namespace = url.Host;
+            return @namespace;
+        }
+
+        /// <summary>
+        /// Get the blob prefix used with EventProcessorHost for a given event hub.
+        /// </summary>
+        /// <param name="eventHubName">the event hub path</param>
+        /// <param name="serviceBusNamespace">the event hub's service bus namespace.</param>
+        /// <returns>a blob prefix path that can be passed to EventProcessorHost.</returns>
+        /// <remarks>
+        /// An event hub is defined by it's path and namespace. The namespace is extracted from the connection string.
+        /// This must be an injective one-to-one function because:
+        /// 1. multiple machines listening on the same event hub must use the same blob prefix. This means it must be deterministic.
+        /// 2. different event hubs must not resolve to the same path.
+        /// </remarks>
+        public static string GetBlobPrefix(string eventHubName, string serviceBusNamespace)
```

Comment:
This isn't accurate for T2, assuming that we're using it to identify checkpoint/ownership.   If not, ignore me.